src/Microsoft.OData.Edm/Csdl/CsdlWriter.cs

```diff
@@ -7,159 +7,141 @@
 using System;
 using System.Collections.Generic;
 using System.Diagnostics;
+using System.IO;
 using System.Linq;
 using System.Xml;
 using Microsoft.OData.Edm.Csdl.Serialization;
+using Microsoft.OData.Edm.Csdl.Json;
 using Microsoft.OData.Edm.Validation;
 
 namespace Microsoft.OData.Edm.Csdl
 {
     /// <summary>
-    /// Provides CSDL serialization services for EDM models.
+    /// Provides CSDL serialization services (XML & JSON) for EDM models.
     /// </summary>
-    public class CsdlWriter
+    public abstract class CsdlWriter

```

Comment:
Changing a concrete class to an abstract class will be a major breaking change. Will this be targetted for 8.x?

Reply:
let me share my thoughts first and  We can talk it more later

1. First, if you check original `CsdlWriter`, the constructor is private, customer can't instantiate it. From this, make abstract is same.

2.   `CsdlWriter` hasn't any public instance methods.

From these, I think it's better to mark it as abstract.